lib/connections/service.go

```diff
@@ -202,27 +202,14 @@ next:
 					continue next
 				}
 
-				// If rate limiting is set, and based on the address we should
-				// limit the connection, then we wrap it in a limiter.
-
-				limit := s.shouldLimit(c.RemoteAddr())
-
-				wr := io.Writer(c)
-				if limit && s.writeRateLimit != nil {
-					wr = NewWriteLimiter(c, s.writeRateLimit)
-				}
-
-				rd := io.Reader(c)
-				if limit && s.readRateLimit != nil {
-					rd = NewReadLimiter(c, s.readRateLimit)
-				}
+				reader, writer := s.setupReaderAndWriterWithOptionalRatelimit(c, s.writeRateLimit, s.readRateLimit, s.cfg.Options())
```

Comment:
This is a "private" method to the service, passing arguments that are part of the service to me it seems redundant.